interop_binaries/src/bin/janus_interop_collector.rs

```diff
@@ -0,0 +1,503 @@
+use anyhow::Context;
+use base64::URL_SAFE_NO_PAD;
+use clap::{Arg, Command};
+use interop_binaries::{
+    install_tracing_subscriber,
+    status::{COMPLETE, ERROR, IN_PROGRESS, SUCCESS},
+    VdafObject,
+};
+use janus_core::{
+    hpke::{
+        self, associated_data_for_aggregate_share, generate_hpke_config_and_private_key,
+        HpkeApplicationInfo, HpkePrivateKey, Label,
+    },
+    message::{
+        Duration, HpkeAeadId, HpkeConfig, HpkeConfigId, HpkeKdfId, HpkeKemId, Interval, Role,
+        TaskId, Time,
+    },
+};
+use janus_server::message::{CollectReq, CollectResp};
+use prio::{
+    codec::{Decode, Encode},
+    field::{Field128, Field64},
+    vdaf::{
+        prio3::{Prio3, Prio3Aes128Count, Prio3Aes128Histogram, Prio3Aes128Sum},
+        AggregateShare, Collector, Vdaf,
+    },
+};
+use reqwest::{
+    header::{CONTENT_TYPE, LOCATION},
+    Url,
+};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::{hash_map::Entry, HashMap},
+    net::{Ipv4Addr, SocketAddr},
+    sync::Arc,
+};
+use tokio::sync::Mutex;
+use warp::{hyper::StatusCode, reply::Response, Filter, Reply};
+
+static DAP_AUTH_TOKEN: &str = "DAP-Auth-Token";
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct AddTaskRequest {
+    task_id: String,
+    leader: String,
+    vdaf: VdafObject,
+    collector_authentication_token: String,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct AddTaskResponse {
+    status: &'static str,
+    #[serde(default)]
+    error: Option<String>,
+    collector_hpke_config: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct CollectStartRequest {
+    task_id: String,
+    agg_param: String,
+    batch_interval_start: u64,
+    batch_interval_duration: u64,
+}
+
+#[derive(Debug, Serialize)]
+struct CollectStartResponse {
+    status: &'static str,
+    #[serde(default)]
+    error: Option<String>,
+    #[serde(default)]
+    handle: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct CollectPollRequest {
+    handle: String,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(untagged)]
+enum AggregationResult {
+    Number(u64),
+    NumberArray(Vec<u64>),
```

Comment:
Is `u64` big enough? `Prio3Aes128Sum` uses `Field128`, so in the end, the result type is `u128`.  Making this generic over `prio::vdaf::Vdaf` would be kind of a pain but since this struct is just used for serializing to JSON, maybe using `u128` instead of `u64` suffices.

Reply:
I specified that the result should be encoded as a JSON number, and `serde_json`'s support for u128 is off by default, and has some warts (due to the limitations of JSON). I'll open an issue on the document to consider changing result types.